kurtosis-devnet/pkg/deploy/fileserver.go

```diff
@@ -67,14 +80,26 @@ func (f *FileServer) Deploy(ctx context.Context, sourceDir string, stateCh <-cha
 	// in order for kurtosis file artifact upload to be idempotent.
 	// (i.e. the file upload and all its downstream dependencies can be SKIPPED on re-runs)
 	tempDir := filepath.Join(baseDir, "upload-content")
-	err = os.Mkdir(tempDir, 0755)
-	if err != nil {
+
+	// Clean up any existing content
+	if exists, _ := afero.Exists(f.fs, tempDir); exists {
+		if err := f.fs.RemoveAll(tempDir); err != nil {
+			return fmt.Errorf("error cleaning up existing directory: %w", err)
+		}
+	}
```

Comment:
nit: do we need to wrap this within an `Exists`? Can't we just try `RemoveAll`?

Reply:
you're absolutely right